### PR TITLE
Style guide

### PR DESCRIPTION
**Pull request by JeffSchering**
Here's the first draft of the proposed style guide. I've broken it up into 4 separate pages. Input welcome! I need to know if I'm on the right track with this.

### CODE REVIEW

File: source/process/sg_grammar-spelling-mechanics.rst [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+Preferred
+  You can view the status in the *Status* pane.
+
+Avoid
+  We'll view the status in the *Status* pane.
+
+Numbers
+=======
+
+Use decimal numbers except when the number is the first word of a sentence. Use commas to make long numbers easier to read.
```
Comment (it33) [inline source/process/sg_grammar-spelling-mechanics.rst]: Hmm... I wasn't clear the first time I read this what a "decimal number" was, and what was the alternative ("written number"?). 

I had the read the examples multiple times and go back to the original description before I figured it out. 

Maybe tweak a bit? 

I think I learned the term "decimal numbers" when I was in grade school, but haven't thought of them since then...

File: source/process/sg_grammar-spelling-mechanics.rst [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+Preferred
+  Three cows ran for 6 kilometers when they saw 2,300,097 mosquitoes chasing them.
+
+Avoid
+  3 cows ran for six kilometers when they saw 2300097 mosquitoes chasing them.
+
+Text highlighting
+=================
+
```
Comment (it33) [inline source/process/sg_grammar-spelling-mechanics.rst]: 1. Should we add an introduction sentence to this section? 
2. Should we add formatting to Highlights section to better communicate how each highlight appears?

File: source/process/sg_mattermost-doc-style.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+This is the Mattermost style guide for documentation. It acts as a reference for writers and editors to ensure that the Mattermost documentation is consistent and clear.
+
+.. Note::
+  The style guide is not intended to slow down or otherwise impede contributions, which are always welcome. No contribution will be rejected due to non-conforming style, although it might be edited.
+
+The Mattermost documentation must be of high quality. It must be accurate and clear, and be presented with a style and tone that is appropriate for technical content. People who use Mattermost rely on the documentation to get their jobs done. We don't want to see an installation of Mattermost delayed because the documentation has an error or is difficult to understand.
+
+.. comments
+
+  Main screen, Navigation panel, Message Details panel
```
Comment (it33) [inline source/process/sg_mattermost-doc-style.rst]: Er... kind of confused by this section...
image

File: source/process/sg_rest_markup.rst [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    
+    a. This is nested item one.
+    b. This is nested item two.
+    
+  3. This is item three.
+  
+Linking to other documents
+==========================
+
+Use relative links to reference other documents in the Mattermost doc set. Also, use either the :doc: or :ref: linking mechanisms that are provided by Sphinx.
```
Comment (it33) [inline source/process/sg_rest_markup.rst]: This section was confusing to me, could we get it to a state where we don't need to go to an external link?

File: source/process/sg_rest_markup.rst [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    _Mattermost Documentation Style Guide: sg_mattermost-doc-style.html
+
+For more information about how to use :doc: and :ref:, see `Inline markup`_ on the Sphinx documentation page.
+
+
+.. _reStructuredText Markup Specification: http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html
+.. _Sphinx Markup Constructs: http://www.sphinx-doc.org/en/stable/markup/index.html
+.. _Inline markup: http://www.sphinx-doc.org/en/stable/markup/inline.html
+
+Literal blocks
```
Comment (it33) [inline source/process/sg_rest_markup.rst]: This section was also hard to follow for me. Perhaps add an intro sentence and some examples so it's more clear to someone who's learning, or coming over form markdown?

File: source/process/sg_rest_markup.rst [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+.. code-block:: none
+
+  .. code-block:: none
+
+
+Menu selections
+===============
+
+To indicate a series of menu selections, avoud the menuselection role that the Sphinx processor provides.
```
Comment (it33) [inline source/process/sg_rest_markup.rst]: It's a little difficult for me to follow "Menu selections", as I think this referring to documentation about selecting menus, but it's appearing in the RST tutorial section. 

My expectation is that the RST tutorial would explain how to make things bold, italic, monospace, etc. and the guideline on how menu items appear would be in a different section?

File: source/process/sg_document-structure.rst [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+Each page should have an introduction that acts as a short description of the document. The short description should be a single paragraph of no more than 3 sentences. Keep in mind that the description is displayed in the search results along with the page title. People read the description to help them decide if the document is the one that they want.
+
+Document sections
+=================
+
+To make pages easier for people to quickly scan for the content that they're looking for, break your document up into logical sections. Each section should have a title, and the title should relate to the content of the section.
+
+A section title is not required if you have only one section.
+
+Avoid subsections. If you find that you need subsections, quite often the document is too long, or too complex, and needs to be broken up into several pages.
```
Comment (JeffSchering) [inline source/process/sg_document-structure.rst]: There are two approaches to this. One is to have one large monolithic file. The other is to have a docset made up of standalone files that are used where needed via `:toctree:`. I suppose there are pros and cons to each approach, but I would advocate for more granular, standalone, topic-based files for all Mattermost docs, not just the style guide, for the following reasons:

1. As a general rule, the content is easier to manage when it's broken up into standalone files with one topic per file. For example, if the MySQL config is one standalone file, it can be pulled into any number of Administration doc sets using `:toctree:` in Sphinx. If something changes and the MySQL instructions need updating, then the change is made in one place and that change automatically propagates to every doc set. If instead the MySQL content is copy/pasted into a dozen long pages, then you need to make sure that you catch every occurrence.
2. Content is easier to find for end users. For example, search the docs for the term `database installation` (https://docs.mattermost.com/search.html?q=database+installation&check_keywords=yes&area=default) and try to figure out which one of the results has your answer. If instead there was a topic called "Installing PostgreSQL database" it would most likely show up in the search results. Furthermore, when you clicked that result, the page would open and the installation instructions would be right there. As it is, none of the search results look promising, so you end up scanning the toc and clicking around until you find something.
	
Having said that, it looks like a lot of open source products are going for the monolithic style these days. I suppose you can make use of the `:include:` directive, which would allow standalone files in the repo and a monolithic file appearing in the published docs. This would leave the options open.

File: source/process/sg_grammar-spelling-mechanics.rst (hunk 1 above)
Comment (JeffSchering) [inline source/process/sg_grammar-spelling-mechanics.rst]: How about: "Spell out numbers when they are the first word in a sentence, otherwise use numeric digits."

File: source/process/sg_grammar-spelling-mechanics.rst (hunk 2 above)
Comment (JeffSchering) [inline source/process/sg_grammar-spelling-mechanics.rst]: Yes... more work needed here.

File: source/process/sg_mattermost-doc-style.rst (hunk 3 above)
Comment (JeffSchering) [inline source/process/sg_mattermost-doc-style.rst]: Those are just comments to myself. They should be removed.

File: source/process/sg_rest_markup.rst (hunk 5 above)
Comment (JeffSchering) [inline source/process/sg_rest_markup.rst]: Makes sense.

File: source/process/sg_rest_markup.rst (hunk 6 above)
Comment (JeffSchering) [inline source/process/sg_rest_markup.rst]: Yes, that makes sense. The reason I have it here is because if you use the `:menuselection:` directive, you end up with normal text and a tiny grey right arrow between items instead of bold text with the `>` character between items.

It's probably worth it to review all the semantic markup elements offered by Sphinx and come up with either a blanket statement about using them or not using them, or specify those which we think should be used.

File: source/process/sg_grammar-spelling-mechanics.rst (hunk 1 above)
Comment (it33) [inline source/process/sg_grammar-spelling-mechanics.rst]: Perfect, thanks!

File: source/process/sg_grammar-spelling-mechanics.rst (hunk 2 above)
Comment (it33) [inline source/process/sg_grammar-spelling-mechanics.rst]: Thanks Jeff,

File: source/process/sg_mattermost-doc-style.rst (hunk 3 above)
Comment (it33) [inline source/process/sg_mattermost-doc-style.rst]: Thanks Jeff,

File: source/process/sg_rest_markup.rst (hunk 4 above)
Comment (it33) [inline source/process/sg_rest_markup.rst]: Thanks Jeff

File: source/process/sg_rest_markup.rst (hunk 5 above)
Comment (it33) [inline source/process/sg_rest_markup.rst]: Thanks Jeff

File: source/process/sg_rest_markup.rst (hunk 6 above)
Comment (it33) [inline source/process/sg_rest_markup.rst]: Thanks Jeff, I think for this phase of the project we can maybe stay with basic RST (just what you see when you read our docs) and we can add the more fancy stuff later? 

Generally knowledge of RST is low, so what you have to start is excellent. We kind of need the 10% of RST that covers 90% of use.

File: source/guides/core.rst [hunk 8]
```diff
@@ -14,8 +14,15 @@ Development Process
...
-   /process/documentation-guidelines*
+   /process/documentation-guidelines.md
 
+Documentation Style Guide
+=========================
+
+.. toctree::
+   :maxdepth: 3
+
+   /process/sg_mattermost-doc-style.rst
```
Comment (it33) [inline source/guides/core.rst]: Sorry if I missed this, just wondering if we should add a note on file naming convention? 

It seems this PR has `sg_` as a prefix, which is different from how other pages are organized. If we use a prefix either we're going to be inconsistent across our pages or we'll have to redirect or break links to refactor everything to use prefixes (which is not bad, so long as we know how to do that properly--what we don't want is orphaned pages). 

Our file naming isn't consistent yet, we should figure out if file naming is in scope for the style guide

File: source/process/sg_document-structure.rst (hunk 7 above)
Comment (it33) [inline source/process/sg_document-structure.rst]: @JeffSchering not sure your thoughts on trying the monolith vs. separate pages? This PR seems to have a number of short and long pages broken up, did you want them still separate?

Comment (JeffSchering) [inline source/process/sg_document-structure.rst]: I kept the pages separate, but when built they're all included into a monolith, which is what readers will see. See sg_mattermost-doc-style.rst for the `..include::` statements.

Here's an idea of what it would look like when ouput to HTML:

image

The *Document sections* stuff is in `sg_document-structure.rst` and *Grammar, spelling, mechanics* is in `sg_grammar-spelling-mechanics.rst`.

File: source/guides/core.rst (hunk 8 above)
Comment (it33) [inline source/guides/core.rst]: Thanks @JeffSchering makes sense, we'll leave file naming out of the guide for now, and we can think about it later--along with a strategy for redirecting out-of-date pages.

File: source/process/sg_document-structure.rst (hunk 7 above)
Comment (it33) [inline source/process/sg_document-structure.rst]: Perfect, thanks @JeffSchering!

**Comment by it33** [comment]
Doc PRs are taking forever, I'm just going to merge this and we can correct anything missed.

**Comment by it33** [comment]
@JeffSchering sorry, I was going to merge this but it looks like there's some conflict, could you tweak and then we'll merge?

**Comment by esethna** [comment]
Hey @JeffSchering, sorry for the delayed review on this, PM team is planning to do a group review tomorrow afternoon.

**Comment by JeffSchering** [comment]
@it33 It should be ok to merge now.

File: source/process/sg_grammar-spelling-mechanics.rst [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+directory name  ``monospace``       ``/opt/mattermost``
+inline code     ``monospace``       ``fmt.Printf("2 times %d = %d\n", x, y )``
+code samples    ``monospace``       See :ref:`syntax-highlight` for an example.
+screen output   ``monospace``       See :ref:`literal-blocks` for an example.
+menu selection  **bold**            "Click **File > Save**."
+UI selection    **bold**            "Click **Next**."
+field names     **bold**            "Enter the font in the **Display Font** field."
+commands        ``monospace``       "At the command line, type ``sudo apt-get install nginx``."
+citations       *italic*            "Read the book *Clean Code* by Robert Martin."
+window titles   *italic*            "The *Account Settings* window opens."
```
Comment (jasonblais) [inline source/process/sg_grammar-spelling-mechanics.rst]: Few tweaks:

1) Please also help add a guideline for
 - keyboard buttons: all capitals with no spaces and no monospace highlighting. Example: CTRL+U
 - placeholder fields: inside square brackets, no other highlighting. Example: [team name]
 - key strokes in monospace: Example: `10`

2) Capitalize the fields under "Text" column (maybe add this part of guidelines for tables)

3) Maybe group `commands` text formatting with the other monospace options. Perhaps below code samples.

4) Propose replacing "menu selection" and "UI selection" with "Clickable Control", which covers both

File: source/process/sg_grammar-spelling-mechanics.rst [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+Avoid
+  3 cows ran for six kilometers when they saw 2300097 mosquitoes chasing them.
+
+Text highlighting
+-----------------
+
+Use highlighting of text to visually set off words and phrases that are important to readers. Content that should be highlighted includes file names, UI controls, and window titles. The following table has a comprehensive list with examples. 
+
+==============  ==================  =======================
+Text            Highlight           Example
```
Comment (jasonblais) [inline source/process/sg_grammar-spelling-mechanics.rst]: Should we add any guidelines for tables? For instance center-aligning the title row in tables?

File: source/process/sg_grammar-spelling-mechanics.rst [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+screen output   ``monospace``       See :ref:`literal-blocks` for an example.
+menu selection  **bold**            "Click **File > Save**."
+UI selection    **bold**            "Click **Next**."
+field names     **bold**            "Enter the font in the **Display Font** field."
+commands        ``monospace``       "At the command line, type ``sudo apt-get install nginx``."
+citations       *italic*            "Read the book *Clean Code* by Robert Martin."
+window titles   *italic*            "The *Account Settings* window opens."
+==============  ==================  =======================
+
+Verb tense
```
Comment (jasonblais) [inline source/process/sg_grammar-spelling-mechanics.rst]: maybe group "Verb tense" with tone and voice earlier in the document?

File: source/process/sg_grammar-spelling-mechanics.rst [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+---------------------
+
+Write documents in English. Use American spelling.
+
+Paragraphs and sentences
+------------------------
+
+Paragraphs should express one idea or topic. Long paragraphs are sometimes difficult to read on screen, so try to keep them to 5 sentences or less. Short paragraphs are easier for people to scan quickly.
+
+Try to keep sentences to 25 words or less in length. Short, single-clause sentences are often easier to understand and easier to translate.
```
Comment (jasonblais) [inline source/process/sg_grammar-spelling-mechanics.rst]: could referencing the [Hemingway writing app](http://www.hemingwayapp.com/) be useful?

It's a common tool to simplify your paragraphs and sentences, which could be helpful for anyone working on docs.

Not sure if you've used the app yourself or if you've used something else before @JeffSchering

File: source/guides/core.rst (hunk 8 above)
Comment (jasonblais) [inline source/guides/core.rst]: We discussed among PMs and propose we remove the `sg_` prefix from the style guide doc file names for now.

The `sg_` prefix will appear in the URL when viewing the page on docs.mattermost.com, and could be confusing if a user doesn't associate `sg_` with `style guide`.

We can consider file naming convention later.

**Comment by esethna** [comment]
@JeffSchering Thanks for this, it's going to be a super useful guide! PM's met to discuss this PR, see the compiled feedback above. Please feel free to ping me if you have any questions!

**Comment by it33** [comment]
@esethna @jasonblais could I ask your help to create a new PR for the proposed changes and manage that among the PM team? 

@JeffSchering's been working on this for a month, we agreed to merge it, 

Regarding `sg_` [this was discussed](https://github.com/mattermost/docs/pull/642#discussion_r89697568) and I believe Jeff is correct that there is no standard for filing naming right now, therefore there is no reason we shouldn't accept a filename with `sg_` . 

Moreover, Jeff is using `sg_` as the prefix for a series of files combined into [one monolithic file](https://docs.mattermost.com/process/sg_mattermost-doc-style.html) with includes, and I personally think that's not a bad system (though we should think about how filenaming plays when a file is included in multiple monolith pages...) 

I think reviews should not be subjective. There should be principles and objective criteria on which reviews are based. [Here is what we have today.](https://docs.mattermost.com/process/sg_mattermost-doc-style.html). 

I'd ask PM's help to update and add to it.  

@JeffSchering huge thanks for starting something the entire community can use!
